packages/monaco/src/browser/monaco-editor-service.ts

```diff
@@ -66,7 +70,8 @@ export class MonacoEditorService extends monaco.services.CodeEditorServiceImpl {
         const mode = this.getEditorOpenMode(input);
         const selection = input.options && this.m2p.asRange(input.options.selection);
         const widgetOptions = this.getWidgetOptions(source, sideBySide);
-        return { mode, selection, widgetOptions };
+        const preview = this.editorPreviewPreferences['editor.enablePreview'];
```

Comment:
I wonder should no we base it on `input.options.pinned` ? See https://github.com/Microsoft/vscode/blob/23fe9191e0a4c3b83068ed70d6613be2b75213ee/src/vs/platform/editor/common/editor.ts#L96-L100 By default it seems to be `false`

so `const preview = !input.options.pinned`

Reply:
it seems great to solve dependency problem if I use !input.options.pinned , I will test later

Reply:
> I wonder should no we base it on `input.options.pinned` ? See https://github.com/Microsoft/vscode/blob/23fe9191e0a4c3b83068ed70d6613be2b75213ee/src/vs/platform/editor/common/editor.ts#L96-L100 By default it seems to be `false`
> 
> so `const preview = !input.options.pinned`
Hi, akosyakov, I found that the IResourceInput has an IResourceInputOptions, but  IResourceInputOptions has no pinned member in this Interface.
`    export interface IResourceInputOptions {
        /**
         * Tells the editor to not receive keyboard focus when the editor is being opened. By default,
         * the editor will receive keyboard focus on open.
         */
        preserveFocus?: boolean;

        /**
         * Will reveal the editor if it is already opened and visible in any of the opened editor groups.
         */
        revealIfVisible?: boolean;

        /**
         * Text editor selection.
         */
        selection?: Partial<monaco.IRange>;
    }`

Reply:
It's there, just not exposed: https://github.com/Microsoft/vscode/blob/6c8f02b41db9ae5c4d15df767d47755e5c73b9d5/src/vs/platform/editor/common/editor.ts#L89-L93

You can add a new property.

Reply:
@akosyakov I've just debugged on the chrome browser, the call stack shows that _openReference function within file:///Users/kosyakov/git/vscode/out-editor-src/vs/editor/contrib/goToDefinition/goToDefinitionCommands.ts only passed the specific options to editorService.openCodeEditor function, so just expose a property named "pinned" will make sense?

```
private _openReference(editor: ICodeEditor, editorService: ICodeEditorService, reference: DefinitionLink, sideBySide: boolean): TPromise<ICodeEditor> {
		const { uri, range } = reference;
		return editorService.openCodeEditor({
			resource: uri,
			options: {
				selection: Range.collapseToStart(range),
				revealIfOpened: true,
				revealInCenterIfOutsideViewport: true
			}
		}, editor, sideBySide);
	}
```
actually, we can find out whether current editor is opened in preview mode by determining 
the parent of 'source' widget is instance of  EditorPreviewWidget, However, It will cause dependency problem as well, therefore, I'd prefer using `const preview = this.editorPreviewPreferences['editor.enablePreview'];`


<img width="1920" alt="_openReference" src="https://user-images.githubusercontent.com/16164849/55671805-5e472300-58c6-11e9-947b-229de8f25425.png">

Reply:
let's do it at least without hard dependency to `@theia/editor-preview`, i.e by using `PreferenceService` instead

Reply:
@akosyakov done